python/ray/_private/services.py

```diff
@@ -384,35 +403,43 @@ def remaining_processes_alive():
     return ray.worker._global_node.remaining_processes_alive()
 
 
-def validate_redis_address(address):
-    """Validates address parameter.
+def canonicalize_bootstrap_address(addr: str):
+    """Canonicalizes Ray cluster bootstrap address to ip:port.
+    Reads address from the environment if needed.
+
+    This function should be used to process user supplied Ray cluster address,
+    via ray.init() or `--address` flags, before using the address to connect.
 
     Returns:
-        redis_address: string containing the full <host:port> address.
-        redis_ip: string representing the host portion of the address.
-        redis_port: integer representing the port portion of the address.
+        Ray cluster address string in <host:port> format.
     """
+    if not addr or addr == "auto":
+        addr = get_ray_address_from_environment()
```

Comment:
`if addr is None or addr == "auto"`

Reply:
I want the condition to include `addr == ""` too.

Reply:
Hmmm, shouldn't this be the case thrown in 419?

if it's `ray.init("")` shouldn't this be treated as an error?